src/reader.rs

```diff
@@ -229,6 +229,9 @@ impl<R: Read + Seek> VdbReader<R> {
         gd: &GridDescriptor,
         count: usize,
     ) -> Result<Vec<T>, ParseError> {
+        if count == 0 {
+            return Ok(vec![T::zeroed(); count]);
```

Comment:
Why construct a `T::zeroed()`... only to have Rust `Copy` it `0` times? (disregarding the optimization that clippy gave you a very strange suggestion for below). If we know `count == 0`, just return an empty vector:


```suggestion
            return Ok(vec![]);
```

Reply:
Ya that didn't make sense. I am constructing an empty vec now.

Reply:
> Ya that didn't make sense. I am constructing an empty vec now.

`vec!` has an optimization that doesnt allocate if the vec is empty iirc

Reply:
Vec::new() does this internally. And vec![] just calls Vec::new():

```
macro_rules! vec {
    () => (
        $crate::__rust_force_expr!($crate::vec::Vec::new())
    );
    ($elem:expr; $n:expr) => (
        $crate::__rust_force_expr!($crate::vec::from_elem($elem, $n))
    );
    ($($x:expr),+ $(,)?) => (
        $crate::__rust_force_expr!(<[_]>::into_vec(
            // This rustc_box is not required, but it produces a dramatic improvement in compile
            // time when constructing arrays with many elements.
            #[rustc_box]
            $crate::boxed::Box::new([$($x),+])
        ))
    );
}
```

Reply:
[`Vec::new()` itself](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.new) doesn't allocate:

> Constructs a new, empty `Vec<T>`.
>
> **The vector will not allocate until elements are pushed onto it.**